test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextTrackOnlyMleNonAsyncApiTests.cs

```diff
@@ -0,0 +1,335 @@
+            Assert.Equal(1, DefaultTrackingContext.Entities.Count);
+            Assert.Equal(1, NonTrackingContext.Entities.Count);
+        }
+
+        private void SaveContextChanges(DataServiceContext[] dataServiceContexts)
+        {
+            foreach (var dataServiceContext in dataServiceContexts)
+            {
+                dataServiceContext.SaveChanges();
+            }
+        }
+
+        private void SetupContextWithRequestPipelineForSaving(DataServiceContext[] dataServiceContexts, bool forMle)
+        {
+            var response = forMle ? DOCUMENT_RESPONSE : USER_RESPONSE;
+            var location = forMle ? "http://localhost:8000/Documents(1)/edit/content" : "http://localhost:8000/Users(1)";
+
+            foreach (var context in dataServiceContexts)
+            {
+                context.Configurations.RequestPipeline.OnMessageCreating =
+                    args => new CustomizedRequestMessage(
+                        args,
+                        response,
+                        new Dictionary<string, string>
+                        {
+                            {"Content-Type", "application/json;charset=utf-8"},
+                            { "Location", location }
+                        });
+            }
+        }
+
+        private void SetupContextWithRequestPipeline(DataServiceContext[] dataServiceContexts, bool forMle)
+        {
+            var response = forMle ? DOCUMENTS_RESPONSE : USERS_RESPONSE;
+
+            foreach (var context in dataServiceContexts)
+            {
+                context.Configurations.RequestPipeline.OnMessageCreating =
+                    args => new CustomizedRequestMessage(
+                        args,
+                        response,
+                        new Dictionary<string, string>
+                        {
+                                {"Content-Type", "application/json;charset=utf-8"}
+                        });
+            }
+        }
+
+        class Container : DataServiceContext
+        {
+            public Container(Uri serviceRoot) :
+                base(serviceRoot, ODataProtocolVersion.V4)
+            {
+                Format.LoadServiceModel = () => CsdlReader.Parse(XmlReader.Create(new StringReader(Edmx)));
+                Format.UseJson();
+                Users = CreateQuery<User>("Users");
+                Documents = CreateQuery<Document>("Documents");
+            }
+            public DataServiceQuery<User> Users { get; private set; }
+            public DataServiceQuery<Document> Documents { get; private set; }
+        }
+    }
+    [Key("Id")]
+    [HasStream]
+    internal class Document : BaseEntityType
+    {
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+        public int FileLength { get; set; }
+    }
+    [Key("Id")]
+    internal class User : BaseEntityType
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+    }

```

Comment:
Would it be worthwile to add a test using a class that does not inherit from `BaseEntityType` to explicitly test what its expected behaviour is?